lib/cloud/gcp/vm.go

```diff
@@ -452,10 +396,22 @@ func (clt *instancesClient) AddSSHKey(ctx context.Context, req *SSHKeyRequest) e
 	if req.PublicKey == nil {
 		return trace.BadParameter("public key not set")
 	}
-	addSSHKey(req.Instance.metadata, req.PublicKey, req.Expires)
+	addSSHKey(req.Instance, req.PublicKey, req.Expires)
+
+	items := make([]*computepb.Items, 0, len(req.Instance.Items))
+	for k, v := range req.Instance.Items {
+		items = append(items, &computepb.Items{Key: googleapi.String(k), Value: googleapi.String(v)})
+	}
+
+	metadata := &computepb.Metadata{
+		Fingerprint: googleapi.String(req.Instance.FingerPrint),
+		Items:       items,
+		Kind:        googleapi.String(req.Instance.Kind),
+	}
```

Comment:
This section repeats in `RemoveSSHKey` and can probably be moved into a helper function.